docs/iam-usage.rst

```diff
@@ -0,0 +1,177 @@
+IAM User Guide
+===================
+
+.. note::
+  This document assumes basic knowledge of Google Cloud IAM,
+  see `the docs <https://cloud.google.com/iam/docs/>`_ for details
+
+IAM Resources
+-------------
+
+The ``iam`` module provides a number of resources necessary for interacting with IAM.
+
+Members
+~~~~~~~
+An IAM member is one of the following:
+
+- ``iam.user(email)`` an individual Google account
+- ``iam.service_account(email)`` a Google Cloud Service Account
+- ``iam.group(email)`` a Google group.
+- ``iam.domain(domain_name)`` a Google apps domain
+- ``iam.ALL_AUTHENTICATED_USERS`` any authenticated Google user
+- ``iam.ALL_USERS``
+
+.. note::
+  that all of these are convenience wrappers around strings.
+  See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+  
+Roles
+~~~~~
+
+Roles represent bundles of permissions that can be added to members.  For a complete list of roles available on a resource run 
+
+``gcloud iam list-grantable-roles //fully/qualified/resource/path``
+
+An ``iam.Role`` object has a name, title, and description
+
+- An ``iam.Role.name`` is the canonical name of a role. This will be the value used as keys in policy dictionaries (see below), and will be referred to as the "role string" throughout this document. E.g. ``'roles/owner'``.
+- ``iam.Role.title`` human readable title of the role. E.g. ``'Owner'``
+- ``iam.Role.description`` the description of a role
+
+Policies
+~~~~~~~~
+
+A policy is a dictionary from role strings to sets of members. ``resource.get_policy()`` also provides etags and versions. An etag is used to provide optimistic concurrency controls on policy updates, while versions are provided for end-user versioning.
+
+
+Policy Changes
+~~~~~~~~~~~~~~
+
+An ``iam.PolicyChange`` object encapsulates changes to made to a policy transactionally. The ``iam.PolicyChange`` constructor takes an optional ``version`` keyword argument, an integer to use as the policy version. If ``version`` is ``None``, when applied ``iam.PolicyChange`` will increment whatever the current version of the policy is by 1.
```

Comment:
> will increment whatever the current version of the policy is by 1.

Does that happen on the client or server side? If on client, won't that require an extra round trip?